File: app/src/main/java/im/point/dotty/model/Comment.java
package im.point.dotty.model;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.annotation.NonNull;

import java.util.Date;

@Entity(tableName = "comments",
        primaryKeys = {"post_id", "id"})
public class Comment {
    @NonNull
    @ColumnInfo(name = "post_id")
    public String postId;
    public int id;
    @ColumnInfo(name = "parent_id")
    public int parentId;
    @ColumnInfo(name = "user_id")
    public long userId;
    public String text;
    public Date timestamp;
}

File: app/src/main/java/im/point/dotty/model/Comment.kt
/*
 * Copyright (c) 2019-2021 Pavel A. Sokolov
 */
package im.point.dotty.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.*

@Entity(tableName = "comments")
data class Comment(
    @PrimaryKey
    val id: String,
    @ColumnInfo(name = "post_id")
    val postId: String,
    @ColumnInfo(name = "user_id")
    val userId: Long,
    val number: Int,
    var login: String,
    @ColumnInfo(name = "reply_to")
    var replyTo: Int? = 0,
    var text: String? = null,
    var timestamp: Date? = null,
    var name: String? = null
) {

    val formattedLogin: String
        get() = "@$login"

    val formattedId: String
        get() = "#$id"
}